### DISCUSSION TITLE
SONOFF SNZB-02D mismatch between reported temperature and displayed temperature

### DISCUSSION THREAD

**Discussion by Slomo17** [Koenkk/zigbee2mqtt#24852] [Devices]
Hello, i noticed a mismatch between what the SONOFF SNZB-02D displays on its display and what is reported over zigbee2mqtt. Its a quiet big mismatch, around 0.5 to 0.7 °C. It' not a Problem fo the reporting interval, the sensor reports frequently but still often after a report the temperatures do not match. Has anyone else observed this problem?

**Comment by DietmarNusch** [comment]
Yes, see that also here for temperature values. Humidity seems to be consistent between display and zigbee2mqtt.

**Reply by Slomo17** [reply]
hm, the question is what is the "correct" value, i think i will have to look for another sensor then, when the sensor shows a different value than the value my automation uses for the heating control, that's just confusing. I guess it's a firmware problem and not a zigbee2mqtt bug

**Comment by chris-1243** [comment]
You need to modify the reporting. By default and in Z2M, this device reports every 1°C change. Mine is set to report every 0.2°C and everything is fine

`Min rep change` 100 = 1°C

**Reply by Slomo17** [reply]
Thanks for the hint, will try this on the weekend, and report back if it solves the problem

**Reply by Slomo17** [reply]
I tried it and it works as you suggested, thank you, this solves my problem.
